runner/utils/driver/istio/apply.go

```diff
@@ -0,0 +1,133 @@
+package istio
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/spaceuptech/space-cloud/runner/model"
+)
+
+// ApplyService deploys the service on istio
+func (i *Istio) ApplyService(_ context.Context, service *model.Service) error {
+	// TODO: do we need to rollback on failure? rollback to previous version if it existed else remove. We also need to rollback the cache in this case
+
+	ns := service.ProjectID
+
+	// Set the default concurrency value to 50
+	if service.Scale.Concurrency == 0 {
+		service.Scale.Concurrency = 50
+	}
+
+	// Adjust the min scale in case of tcp based services. Min scale for tcp services need to be at least 1.
+	adjustMinScale(service)
+
+	// TODO: remove artifact store related code
+	token, err := i.auth.GenerateTokenForArtifactStore(service.ID, service.ProjectID, service.Version)
+	if err != nil {
+		return err
```

Comment:
log the errors here using logrus

Reply:
I'm logging the error here because it eventually gets logged in the handler